include/zephyr/drivers/timer/ti_dmtimer.h

```diff
@@ -0,0 +1,158 @@
+/* Copyright (C) 2023 BeagleBoard.org Foundation
+ * Copyright (C) 2023 S Prashanth
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_DRIVERS_TIMERS_TI_DMTIMER_H_
+#define ZEPHYR_DRIVERS_TIMERS_TI_DMTIMER_H_
+
+#include <zephyr/devicetree.h>
+
+#define TI_DM_TIMER_BASE_ADDR      DT_REG_ADDR(DT_INST(0, ti_am654_dmtimer))
+
+#define TI_DM_TIMER_IRQ_NUM        DT_IRQN(DT_INST(0, ti_am654_dmtimer))
+#define TI_DM_TIMER_IRQ_PRIO       DT_IRQ(DT_INST(0, ti_am654_dmtimer), priority)
+#define TI_DM_TIMER_IRQ_FLAGS      DT_IRQ(DT_INST(0, ti_am654_dmtimer), flags)
+
+#define TI_DM_TIMER_TIDR           (TI_DM_TIMER_BASE_ADDR + 0x00)
+#define TI_DM_TIMER_TIOCP_CFG      (TI_DM_TIMER_BASE_ADDR + 0x10)
+#define TI_DM_TIMER_IRQ_EOI        (TI_DM_TIMER_BASE_ADDR + 0x20)
+#define TI_DM_TIMER_IRQSTATUS_RAW  (TI_DM_TIMER_BASE_ADDR + 0x24)
+#define TI_DM_TIMER_IRQSTATUS      (TI_DM_TIMER_BASE_ADDR + 0x28)
+#define TI_DM_TIMER_IRQENABLE_SET  (TI_DM_TIMER_BASE_ADDR + 0x2c)
+#define TI_DM_TIMER_IRQENABLE_CLR  (TI_DM_TIMER_BASE_ADDR + 0x30)
+#define TI_DM_TIMER_IRQWAKEEN      (TI_DM_TIMER_BASE_ADDR + 0x34)
+#define TI_DM_TIMER_TCLR           (TI_DM_TIMER_BASE_ADDR + 0x38)
+#define TI_DM_TIMER_TCRR           (TI_DM_TIMER_BASE_ADDR + 0x3c)
+#define TI_DM_TIMER_TLDR           (TI_DM_TIMER_BASE_ADDR + 0x40)
+#define TI_DM_TIMER_TTGR           (TI_DM_TIMER_BASE_ADDR + 0x44)
+#define TI_DM_TIMER_TWPS           (TI_DM_TIMER_BASE_ADDR + 0x48)
+#define TI_DM_TIMER_TMAR           (TI_DM_TIMER_BASE_ADDR + 0x4c)
+#define TI_DM_TIMER_TCAR1          (TI_DM_TIMER_BASE_ADDR + 0x50)
+#define TI_DM_TIMER_TSICR          (TI_DM_TIMER_BASE_ADDR + 0x54)
+#define TI_DM_TIMER_TCAR2          (TI_DM_TIMER_BASE_ADDR + 0x58)
+#define TI_DM_TIMER_TPIR           (TI_DM_TIMER_BASE_ADDR + 0x5c)
+#define TI_DM_TIMER_TNIR           (TI_DM_TIMER_BASE_ADDR + 0x60)
+#define TI_DM_TIMER_TCVR           (TI_DM_TIMER_BASE_ADDR + 0x64)
+#define TI_DM_TIMER_TOCR           (TI_DM_TIMER_BASE_ADDR + 0x68)
+#define TI_DM_TIMER_TOWR           (TI_DM_TIMER_BASE_ADDR + 0x6c)
+
```

Comment:
Precalculating the register addresses for a single hardcoded instance seems not scalable especially when multiple timer instances are present, as discussed in the DM timer PR I will recommend fixing this.

Reply:
Sure, will go fix that in the DM timer PR (and rebase so the fix ends up here too).